src/high-level/tests/KinDynComputationsMatrixViewAndSpanUnitTest.cpp

```diff
@@ -47,37 +49,39 @@ int real_random_int(int initialValue, int finalValue)
 void setRandomState(iDynTree::KinDynComputations & dynComp)
 {
     size_t dofs = dynComp.getNrOfDegreesOfFreedom();
+    size_t posCoords = dynComp.model().getNrOfPosCoords();
     Transform    worldTbase;
     Eigen::Vector6d baseVel;
     Eigen::Vector3d gravity;
 
-    Eigen::VectorXd qj(dofs), dqj(dofs), ddqj(dofs);
+    Eigen::VectorXd qj(posCoords), dqj(dofs), ddqj(dofs);
 
-    worldTbase = iDynTree::Transform(Rotation::RPY(random_double(),random_double(),random_double()),
-                                     Position(random_double(),random_double(),random_double()));
+    // Use utility functions for generating random state
+    worldTbase = getRandomTransform();
 
+    // Generate random base velocity using iDynTree utility and convert to Eigen
+    Twist baseTwist = getRandomTwist();
+    baseVel = toEigen(baseTwist);
 
-    Eigen::Matrix4d transform = toEigen(worldTbase.asHomogeneousTransform());
+    // Generate random gravity vector
+    Vector3 gravityVec;
+    getRandomVector(gravityVec);
+    gravity = toEigen(gravityVec);
+    gravity(2) = 0.0;  // Keep the original constraint
```

Comment:
```suggestion
```

the old line seems just a leftover, and the `// Keep the original constraint` comment does not make any sense. Let's just remove this.